### DISCUSSION TITLE
(cdk): Support for infrastructure and code stacks as peer stacks

### DISCUSSION THREAD

**Discussion by SamStephens** [aws/aws-cdk#20853] [General]
### Describe the issue

I'm trying to use the classic pattern where you have an infrastructure stack and a code stack. I want to define these two stacks as peer stacks, and directly reference resources in the infrastructure stack from the code stack. When I started doing this, I quickly encountered a [cyclic reference error that I've opened a separate bug issue for](https://github.com/aws/aws-cdk/issues/20256).

However I'm wanting to know whether this is a usage pattern that the CDK team support in general, and whether the CDK consider this poor practice or good practice; before I commit to it and then potentially find I'm stuck because I'm using the CDK in an unsupported fashion.

I'm not interested in using nested stacks for this. When I worked with Nested Stacks back when I was an AWS employee I found them quite clunky to work with, although this was a few years ago.

I'm currently using peer stacks, with the infrastructure stack exporting resource ARNs using [CfnOutput](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.CfnOutput.html), and the code stack then importing them using [Fn.importValue](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.Fn.html#static-importwbrvaluesharedvaluetoimport). But it would be cleaner if I could safely pass the actual CDK constructs from the infrastructure stack to the code stack.

### Links

https://docs.aws.amazon.com/cdk/v2/guide/stacks.html

**Comment by indrora** [comment]
The common pattern with CDK is that you keep the resources (assets, code, etc.) that are relevant to a specific area of your broader infrastructure within one stack. 

This could be one service, one aspect of a service, or some other discrete unit of deployment.
